pydantic/_internal/_generate_schema.py

```diff
@@ -568,7 +579,9 @@ def _namedtuple_schema(self, namedtuple_cls: Any) -> core_schema.CallSchema:
         """
         Generate schema for a NamedTuple.
         """
-        annotations: dict[str, Any] = _typing_extra.get_type_hints(namedtuple_cls, include_extras=True)
+        annotations: dict[str, Any] = get_type_hints_infer_globalns(
+            namedtuple_cls, include_extras=True, localns=self.types_namespace
```

Comment:
It seems like `localns` is always `self.types_namespace`. Should we rename `self.types_namespace` to `localns` elsewhere? Or should we have some sort of explanation in the docstrings of `GenerateSchema` explaining what it expects `types_namespace` to be and how you should get it (`sys._getframe(1).f_globals.copy()` of wherever you are called?).

Reply:
probably; I haven't really looked into how it gets used, when wrestling with generics it has mostly "just worked" for me. I think when we get generic dataclasses working and/or when we refactor generics to drop the contextvar, it will clear things up a bit.